src/request_parser.ts

```diff
@@ -36,9 +35,20 @@ export const parseTakerRequest = (req: Pick<express.Request, 'headers' | 'query'
             comparisonPrice: query.comparisonPrice ? new BigNumber(query.comparisonPrice) : undefined,
         };
 
-        const takerRequest: TakerRequest = query.sellAmountBaseUnits
-            ? { ...takerRequestBase, sellAmountBaseUnits: new BigNumber(query.sellAmountBaseUnits) }
-            : { ...takerRequestBase, buyAmountBaseUnits: new BigNumber(query.buyAmountBaseUnits) };
+        let takerRequest: TakerRequest;
+        if (query.sellAmountBaseUnits) {
+            takerRequest = {
+                ...takerRequestBase,
+                sellAmountBaseUnits: new BigNumber(query.sellAmountBaseUnits),
+            };
+        } else if (query.buyAmountBaseUnits) {
+            takerRequest = {
+                ...takerRequestBase,
+                buyAmountBaseUnits: new BigNumber(query.buyAmountBaseUnits),
+            };
+        } else {
+            throw new Error('"buyAmountBaseUnits" and "sellAmountBaseUnits" are mutually exclusive');
```

Comment:
I think this error message can be clarified.  To me, "mutually exclusive" means, "you can't specify both." But the case you're catching here is if the client hasn't specified *either*, not both, right?  So maybe something like `a request must specify either a buyAmountBaseUnits or a sellAmountBaseUnits`, and if you'd like maybe also add `... (but not both)`.  Technically here you wouldn't catch the case of  them specifying both, because it would just  be handled by the first branch of the `if`, right?  So if you really care about enforcing mutual exclusivity then you could add some more logic here.